numpy_questions.py

```diff
@@ -29,39 +29,51 @@ def max_index(X):
     Returns
     -------
     (i, j) : tuple(int)
-        The row and columnd index of the maximum.
+        The row and column index of the maximum.
 
     Raises
     ------
     ValueError
         If the input is not a numpy array or
         if the shape is not 2D.
     """
-    i = 0
-    j = 0
+    if not isinstance(X, np.ndarray):
+        raise ValueError("Input must be a numpy array.")
+    if X.ndim != 2:
+        raise ValueError("Input array must be 2D.")
 
-    # TODO
+    max_value = X[0, 0]
+    max_position = (0, 0)
 
-    return i, j
+    for i in range(X.shape[0]):
+        for j in range(X.shape[1]):
+            if X[i, j] > max_value:
+                max_value = X[i, j]
+                max_position = (i, j)
+
+    return max_position
 
 
 def wallis_product(n_terms):
     """Implement the Wallis product to compute an approximation of pi.
 
-    See:
-    https://en.wikipedia.org/wiki/Wallis_product
-
     Parameters
     ----------
     n_terms : int
-        Number of steps in the Wallis product. Note that `n_terms=0` will
-        consider the product to be `1`.
+        Number of steps in the Wallis product.
```

Comment:
@IliassSjm why does this change?

Reply:
Opps i must have cut it instead of copying it